mgmtsystem_audit/models/mgmtsystem_audit.py

```diff
@@ -20,163 +20,265 @@
+        'mgmtsystem_audit_imp_opp_rel',
+        'mgmtsystem_action_id',
+        'mgmtsystem_audit_id',
+        'Improvement Opportunities',
+    )
+
+    nonconformity_ids = fields.Many2many(
+        'mgmtsystem.nonconformity',
+        string='Nonconformities',
+    )
+    state = fields.Selection(
+        [
+            ('open', 'Open'),
+            ('done', 'Closed'),
+        ],
+        'State',
+        default="open"
+    )
+    system_id = fields.Many2one('mgmtsystem.system', 'System')
+    company_id = fields.Many2one(
+        'res.company', 'Company', default=_own_company)
+
+    @api.depends("nonconformity_ids")
+    def _compute_number_of_nonconformities(self):
+        """Count number of nonconformities."""
+        number = 0
+        for id in self.nonconformity_ids:
+            number = number + 1
+        self.number_of_nonconformities = number
+        return number
+
+    @api.depends("imp_opp_ids")
+    def _compute_number_of_improvement_opportunities(self):
+        """Count number of improvements Opportunities."""
+        number = 0
+        for id in self.imp_opp_ids:
+            number = number + 1
+        self.number_of_improvements_opportunity = number
+        return number
+
+    @api.depends("line_ids")
+    def _compute_number_of_questions_in_verification_list(self):
+        number = 0
+        for id in self.line_ids:
+            number = number + 1
+        self.number_of_questions_in_verification_list = number
+        return number
+
+    @api.depends("write_date")
+    def _compute_days_since_last_update(self):
+        for audit in self:
+            audit.days_since_last_update = audit._elapsed_days(
+                audit.create_date,
+                audit.write_date)
+
+    @api.model
+    def _elapsed_days(self, dt1_text, dt2_text):
+        res = 0
+        if dt1_text and dt2_text:
+            dt1 = fields.Datetime.from_string(dt1_text)
+            dt2 = fields.Datetime.from_string(dt2_text)
+            res = (dt2 - dt1).days
+        return res
+
+    @api.model
+    def create(self, vals):
+        """Audit creation."""
         vals.update({
-            'reference': self.pool.get('ir.sequence').get(
-                cr, uid, 'mgmtsystem.audit'
-            )
+            'reference': self.env['ir.sequence'].next_by_code(
+                'mgmtsystem.audit'
+            ),
         })
-        return super(mgmtsystem_audit, self).create(
-            cr, uid, vals, context=context)
+        audit_id = super(MgmtSystemAudit, self).create(vals)
+        return audit_id
 
-    def button_close(self, cr, uid, ids, context=None):
+    @api.multi
+    def button_close(self):
         """When Audit is closed, post a message to followers' chatter."""
-        self.message_post(cr, uid, ids, _("Audit closed"), context=context)
-        return self.write(cr, uid, ids, {'state': 'done'})
+        self.message_post(_("Audit closed"))
+        number_of_days_to_close = (
```

Comment:
Have number to days to close as a computed field.
Just write on the closed date field.